### PR TITLE
Introduce `sourcemapIgnoreList` config predicate

### PR DESCRIPTION
**Pull request by bmeurer**
This PR contains:

- [ ] bugfix
- [x] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

List any relevant issue numbers:

#4847

### Description

This PR updates rollup to leverage the initial `x_google_ignoreList` support in magic-string, and introduces a `sourcemapIgnoreList` config predicate that works similar to the existing `sourcemapPathTransform`, and can be provided to mark certain source files as ignore-listed when generating source maps.

This is useful for libraries and frameworks that provide code that is not relevant to the debugging experience of the customer. Chrome DevTools already supports this new `x_google_ignoreList` field in the source maps[^1].

Doc: https://goo.gle/devtools-ignoreList-adoption

[^1]: https://developer.chrome.com/blog/devtools-better-angular-debugging

### CODE REVIEW

**Comment by bmeurer** [comment]
I rebased this on the `magic-string` update. This is ready for review now.

File: src/utils/renderChunks.ts [hunk 1]
```diff
@@ -158,21 +159,31 @@ async function transformChunk(
...
-			.map(normalize);
+			}
+			if (sourcemapPathTransform) {
+				sourcePath = sourcemapPathTransform(sourcePath, sourcemapPath);
+				if (typeof sourcePath !== 'string') {
+					error(errorFailedValidation(`sourcemapPathTransform function must return a string.`));
+				}
+			}
+			map.sources[sourcesIndex] = normalize(sourcePath);
+		}
```
Comment (lukastaegert) [inline src/utils/renderChunks.ts:186]: That looks like it should definitely not hurt performance, thanks a lot!

**Review by lukastaegert — APPROVED** [review]
Great stuff! Especially considering there was virtually not feedback loop you picked up nicely how to write Rollup tests etc ;)

Only minor point for me is that the tests never look at the second argument of the sourcemapIgnoreList function, and only vaguely at the first one. Considering these are direct parts of the API, it would be nice at least at one point to assert what is passed here to avoid breaking changes if we should ever refactor here.

Otherwise, you got nearly 100% coverage :) The only thing missing is the else case for `if (!map.x_google_ignoreList.includes(sourcesIndex)) {`, but if I understand correctly, this is probably close to impossible to encounter while on the other side it would be foolish not to handle this case explicitly to avoid future regressions. So this is fine from my side.

**Comment by bmeurer** [comment]
> Only minor point for me is that the tests never look at the second argument of the sourcemapIgnoreList function, and only vaguely at the first one. Considering these are direct parts of the API, it would be nice at least at one point to assert what is passed here to avoid breaking changes if we should ever refactor here.

I added a trivial test case that also sanity checks the second parameter.

> Otherwise, you got nearly 100% coverage :) The only thing missing is the else case for if (!map.x_google_ignoreList.includes(sourcesIndex)) {, but if I understand correctly, this is probably close to impossible to encounter while on the other side it would be foolish not to handle this case explicitly to avoid future regressions. So this is fine from my side.

I'm planning to test this, once rollup preserves `x_google_ignoreList` from input source maps.

**Review by lukastaegert — APPROVED** [review]

**Pull request merged**